torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,416 @@
+import os
+import re
+import torch
+from torchtext.utils import download_from_url, extract_archive
+from torchtext.data.utils import generate_ngrams
+from torchtext.vocab import build_vocab
+import random
+from torchtext import data
+from tqdm import tqdm
+from torchtext.data import dataset
+from torchtext.data.iterator import generate_iterators
+
+
+def download(url, raw_folder, dataset_name):
+    """Download the dataset if it doesn't exist in processed_folder already."""
+
+    train_csv_path = os.path.join(raw_folder,
+                                  dataset_name + '_csv',
+                                  'train.csv')
+    test_csv_path = os.path.join(raw_folder,
+                                 dataset_name + '_csv',
+                                 'test.csv')
+    if os.path.exists(train_csv_path) and os.path.exists(test_csv_path):
+        return
+
+    os.makedirs(raw_folder)
+    filename = dataset_name + '_csv.tar.gz'
+    url = url
+    path = os.path.join(raw_folder, filename)
+    download_from_url(url, path)
+    extract_archive(path, raw_folder, remove_finished=True)
+
+    print('Dataset %s downloaded.' % dataset_name)
+
+
+def text_normalize(line):
+    """Normalize text string and separate label/text."""
+
+    line = line.lower()
+    label, text = line.split(",", 1)
+    label = "__label__" + re.sub(r'[^0-9\s]', '', label)
+    text = re.sub(r'[^a-zA-Z0-9\s]', ' ', text)
+    text = ' '.join(text.split())
+    line = label + ' , ' + text + ' \n'
+    return line
+
+
+def preprocess(raw_folder, processed_folder, dataset_name):
+    """Preprocess the csv files."""
+
+    raw_folder = os.path.join(raw_folder, dataset_name.lower() + '_csv')
+
+    if os.path.exists(processed_folder) is not True:
+        os.makedirs(processed_folder)
+
+    src_filepath = os.path.join(raw_folder, 'train.csv')
+    tgt_filepath = os.path.join(processed_folder, dataset_name + '.train')
+    lines = []
+    with open(src_filepath) as src_data:
+        with open(tgt_filepath, 'w') as new_data:
+            for line in src_data:
+                lines.append(text_normalize(line))
+            random.shuffle(lines)
+            new_data.writelines(lines)
+
+    src_filepath = os.path.join(raw_folder, 'test.csv')
+    tgt_filepath = os.path.join(processed_folder, dataset_name + '.test')
+    lines = []
+    with open(src_filepath) as src_data:
+        with open(tgt_filepath, 'w') as new_data:
+            for line in src_data:
+                lines.append(text_normalize(line))
+            random.shuffle(lines)
+            new_data.writelines(lines)
+
+    print("Dataset %s preprocessed." % dataset_name)
+
+
+def load_text_classification_data(filepath, fields, ngrams=1):
```

Comment:
I'd add an underscore ("_load_text[...]"), because this function is very specific to these datasets and as such a helper.